### PR TITLE
update recipe for 3.12 rebuild

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
pytest-benchmark 4.0.0

**Destination channel:** defaults

### Links

- [PKG-4495](https://anaconda.atlassian.net/browse/PKG-4495) 
- [Upstream repository](https://github.com/ionelmc/pytest-benchmark)
- [Upstream changelog/diff](https://github.com/ionelmc/pytest-benchmark/releases/tag/v4.0.0)
- Relevant dependency PRs:
  - graphql-core | https://cloud.prefect.io/anaconda/flow-run/927f2d4b-20f2-4833-a11d-df077164aba8?logs=&logId=1ac95ce3-491d-49e3-918f-7afe8eb9af79

### Explanation of changes:
- increase build number and add description

[PKG-4495]: https://anaconda.atlassian.net/browse/PKG-4495?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,18 +10,20 @@ source:
   sha256: fb0785b83efe599a6a956361c0691ae1dbb5318018561af10f3e915caa0048d1
 
 build:
+  skip: true # [py>37]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
  skip: true # [py<37]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,18 +10,20 @@ source:
...
+  number: 2
 
 requirements:
   host:
     - python
-    - setuptools
+    - setuptools >=30.3.0
     - wheel
     - pip
+    - py
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:26]: Is it needed in `host` for building the package?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,18 +10,20 @@ source:
...
   run:
     - python
     - pytest >=3.8
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:29]: Seems `py` was removed in `v4.0.0` https://github.com/ionelmc/pytest-benchmark/commit/c2e860f628116473e3134aba21492c04650d00a0

File: recipe/meta.yaml (hunk 2 above)
Comment (JamesRobertsonGames) [reply recipe/meta.yaml:26]: Linter was keen on it being here in this instance, but happy to remove on recommendation?

File: recipe/meta.yaml (hunk 3 above)
Comment (JamesRobertsonGames) [reply recipe/meta.yaml:29]: Thought the same but kept it there due to the conversation in a previous PR
https://github.com/AnacondaRecipes/pytest-benchmark-feedstock/pull/3#discussion_r1368320862

**Review by SC426 — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**